src/main/java/ui/components/pickers/MilestonePickerDialog.java

```diff
@@ -280,6 +294,12 @@ private HBox createPreviouslyAssignedMilestoneBox() {
                 .findAny();
     }
 
+    private Optional<PickerMilestone> getNextOpenMilestone(List<PickerMilestone> milestoneList) {
```

Comment:
This should be a `static` method in `PickerMilestone` class. The method signature can remain the same. Put header comment about what this method does, especially that the definition of next open milestone is not widely known (some people might think it refers to the closest milestone, others the next milestone after closest).

Also, I am guessing this method requires the input list to be sorted. Put in the comment, e.g. `Condition: input list must be sorted according to <...>`.